ProChess/Shared/Rules/InCheckPossibility.cs

```diff
@@ -0,0 +1,380 @@
+                if ((opponentSide != null) || (kingSide != null))
+                    return false;
+            }
+            return false;
+        }
+        private bool BishopAndQueenAttackTopRight(Piece theKingPiece, List<Piece> kingSidePieces, List<Piece> opponentSidePieces)
+        {
+            var column = theKingPiece.StartColumn;
+
+            for (var row = theKingPiece.StartRow + _directionOffsets[0]; row <= _edgeBoard[1]; row++)
+            {
+                column++;
+                if (column > _edgeBoard[1])
+                    break;   
+
+                Piece? opponentSide = opponentSidePieces.FirstOrDefault(x => x.StartRow == row && x.StartColumn == column);
+                Piece? kingSide = kingSidePieces.FirstOrDefault(x => x.StartRow == row && x.StartColumn == column);
+
+                if ((opponentSide as Bishop != null) || (opponentSide as Queen != null))
+                    return true;
+                if ((opponentSide != null) || (kingSide != null))
+                    return false;
+            }
+            return false;
+        }
+        private bool BishopAndQueenAttackBottomLeft(Piece theKingPiece, List<Piece> kingSidePieces, List<Piece> opponentSidePieces)
+        {
+            var column = theKingPiece.StartColumn;
+
+            for (var row = theKingPiece.StartRow + _directionOffsets[1]; row >= _edgeBoard[0]; row--)
+            {
+                column--;
+                if (column > _edgeBoard[1])
+                    break;
+
+                Piece? opponentSide = opponentSidePieces.FirstOrDefault(x => x.StartRow == row && x.StartColumn == column);
+                Piece? kingSide = kingSidePieces.FirstOrDefault(x => x.StartRow == row && x.StartColumn == column);
+
+                if ((opponentSide as Bishop != null) || (opponentSide as Queen != null))
+                    return true;
+                if ((opponentSide != null) || (kingSide != null))
+                    return false;
+            }
+            return false;
+        }
+        private bool BishopAndQueenAttackBottomRight(Piece theKingPiece, List<Piece> kingSidePieces, List<Piece> opponentSidePieces)
+        {
+            var column = theKingPiece.StartColumn;
+
+            for (var row = theKingPiece.StartRow + _directionOffsets[1]; row >= _edgeBoard[0]; row--)
+            {
+                column++;
+                if (column > _edgeBoard[1])
+                    break;
+
+                Piece? opponentSide = opponentSidePieces.FirstOrDefault(x => x.StartRow == row && x.StartColumn == column);
+                Piece? kingSide = kingSidePieces.FirstOrDefault(x => x.StartRow == row && x.StartColumn == column);
+
+                if ((opponentSide as Bishop != null) || (opponentSide as Queen != null))
+                    return true;
+                if ((opponentSide != null) || (kingSide != null))
+                    return false;
+            }
+            return false;
+        }
+    }
+}
```

Comment:
I get the impression that you can use the implementation of the rules. This way we don't duplicate code